p4include/v1model.p4

```diff
@@ -140,6 +135,27 @@ extern void clone3<T>(in CloneType type, in bit<32> session, in T data);
 
 extern void truncate(in bit<32> length);
 
+/**
+Verifies the 16-bit checksum of the supplied data.
+T must be a list expression where all the fields are bit-fields or varbits
+and where the total dynamic length of the fields is a multiple of 16 bits.
+If this method detects that a checksum of the data is not correct it
+sets an internal error flag.
+@param condition  If 'false' the verification always succeeds.
+@param data       Data whose checksum is verified.
+@param checksum   Expected checksum of the data.
+*/
+extern void verify_checksum<T>(in bool condition, in T data, in bit<16> checksum);
```

Comment:
I think that was my question. Is there a reason not to support the other algorithms in the backend? It looks like you would just need to map the enum value to the algo name and call `createCalculation` with that name as the first argument. It may be worth it to add support now rather than modify v1model.p4 in a non-backward compatible fashion in the future.